src/community/features-templating/features-templating-ows/src/main/java/org/geoserver/featurestemplating/ows/wfs/TemplateGetFeatureResponseHelper.java

```diff
@@ -70,7 +70,7 @@ TemplateOutputWriter getOutputWriter(OutputStream output, String version) throws
             case GML31:
             case GML32:
             case HTML:
-                XMLOutputFactory xMLOutputFactory = XMLOutputFactory.newInstance();
+                XMLOutputFactory xMLOutputFactory = XMLOutputFactory.newDefaultFactory();
```

Comment:
@Mike7311 can you elaborate on this change (how and why the change works).

I'm also noting that the [newDefaultFactory method](https://docs.oracle.com/en/java/javase/11/docs/api/java.xml/javax/xml/stream/XMLOutputFactory.html#newDefaultFactory()) is available since Java 9, so the change will work on main and 2.23.x, but won't compile back on 2.22.x and 2.21.x.

Which of course may be a problem, or not, depending on what backport targets you have.

Reply:
Of course. newInstance() finds some implementation of XMLOutputFactory, but in this case this implementation didn't support ESCAPE_CHARS property, which is set on the next lines, thus making factory escape special characters.
![image](https://user-images.githubusercontent.com/29422783/230372332-422bcf83-30e7-44a7-9316-ce3732ec9310.png)

Reply:
As for backtracking I will discuss it

Reply:
Changed factory for java 8 support